src/libsystemd/sd-event/sd-event.c

```diff
@@ -3777,8 +3777,6 @@ static int event_prepare(sd_event *e) {
 
                 s->prepare_iteration = e->iteration;
                 r = prioq_reshuffle(e->prepare, s, &s->prepare_index);
-                if (r < 0)
-                        return r;
```

Comment:
well, that might be the case, but then the `r =` assignment should also go.

Reply:
Yes, let me check all the use cases and submit a new one.